packages/location-management/src/components/LocationUnitList/index.tsx

```diff
@@ -103,35 +103,25 @@ export const LocationUnitList: React.FC<Props> = (props: Props) => {
     }
   );
 
-  useQueries(
+  const treeDataQuery = useQueries(
     locationUnits.data
       ? locationUnits.data.map((location) => {
           return {
             queryKey: [LOCATION_HIERARCHY, location.id],
             queryFn: () => new OpenSRPService(LOCATION_HIERARCHY, opensrpBaseURL).read(location.id),
             onError: () => sendErrorNotification(lang.ERROR_OCCURRED),
             // Todo : useQueries doesn't support select or types yet https://github.com/tannerlinsley/react-query/pull/1527
-            onSuccess: (res) => {
-              // the Tree is not already in the locationtree
-              if (!treeData.find((data) => JSON.stringify(data) === JSON.stringify(res))) {
-                // if the tree already exist i.e only a part of tree is changed, we need to replce old one else add new one
-                const alreadyExist = treeData.findIndex(
-                  (tree) => tree.id === (res as ParsedHierarchyNode).id
-                );
-
-                if (alreadyExist !== -1)
-                  treeData.splice(alreadyExist, 1, res as ParsedHierarchyNode);
-                else setTreeData((treeData) => [...treeData, res as ParsedHierarchyNode]);
-              }
-            },
-            // Todo : useQueries doesn't support select or types yet https://github.com/tannerlinsley/react-query/pull/1527
             select: (res) => generateJurisdictionTree(res as RawOpenSRPHierarchy).model,
           };
         })
       : []
-  );
+  ) as UseQueryResult<ParsedHierarchyNode>[];
+
+  const treeData = treeDataQuery
+    .map((query) => query.data)
+    .filter((e) => e !== undefined) as ParsedHierarchyNode[];
 
-  useEffect(() => {
+  useDeepCompareEffect(() => {
```

Comment:
We could use JSON.stringify to deep compare the objects and thus avoid useDeepCompareEffect